xpdan/data_reduction.py

```diff
@@ -172,107 +173,203 @@ def _npt_cal(config_dict, total_shape=(2048, 2048)):
+        addtional keywords to overwrite integration behavior. Please
+        refer to pyFAI.azimuthalIntegrator.AzimuthalIntegrator for
+        more information
+
+    Note
+    ----
+    complete docstring of masking functionality could be find in
+    ``mask_img``
+
+    customized mask can be assign to by kwargs (It must be a ndarray)
+    >>> integrate_and_save(mask=my_mask)
+
+    See also
+    --------
+    xpdan.tools.mask_img
+    pyFAI.azimuthalIntegrator.AzimuthalIntegrator
     """
     # normalize list
     header_list = _prepare_header_list(headers)
 
     # config_dict
     if config_dict is None:
-        config_dict = _load_config()  # default one
+        config_dict = _load_config() # default dict 
+
+    # setting up geometry
     ai.setPyFAI(**config_dict)
     npt = _npt_cal(config_dict)
 
+    total_rv_list_Q = []
+    total_rv_list_2theta = []
+
     # iterate over header
-    total_rv_list = []
     for header in header_list:
         root = header.start.get(handler.root_dir_name, None)
         if root is not None:
             root_dir = os.path.join(W_DIR, root)
             os.makedirs(root_dir, exist_ok=True)
         else:
             root_dir = W_DIR
-        header_rv_list = []
+        header_rv_list_Q = []
+        header_rv_list_2theta = []
         # dark logic
-        dark_img = handler.pull_dark(header)
+        dark_img = None
+        if dark_sub:
+            dark_img, dark_time = handler.pull_dark(header)
         for event in handler.exp_db.get_events(header, fill=True):
+            # dark subtraction
             img, event_timestamp, ind, dark_sub = handler._dark_sub(event,
                                                                     dark_img)
+            # basic file name
             f_name = handler._file_name(event, event_timestamp, ind)
             if dark_sub:
                 f_name = 'sub_' + f_name
-            stem, ext = os.path.splitext(f_name)
-            chi_name = stem + '.chi'
-            integration_dict = {'filename': os.path.join(root_dir, chi_name),
-                                'polarization_factor':
-                                    polarization_factor}
-            print("INFO: integrating image: {}".format(f_name))
-            rv = ai.integrate1d(img, npt, **integration_dict)
-            header_rv_list.append(rv)
-            print("INFO: save chi file: {}".format(chi_name))
-        total_rv_list.append(header_rv_list)
-        # each header generate  a list of rv
 
-    print(" *** {} *** ".format('Integration process finished'))
-    print("INFO: chi files are saved at {}".format(root_dir))
-    return total_rv_list
+            # masking logic
+            mask = None
+            if auto_mask:
+                print("INFO: mask your image: {}".format(f_name))
+                f_name = 'masked_' + f_name
+                if mask_dict is None:
+                    mask_dict = an_glbl.mask_dict
+                mask = mask_img(img, ai, **mask_dict)
```

Comment:
The polarization correction is not for the integration, at least not directly. The image needs to have it's pixels have the smallest standard deviation possible, thus we need to correct for anything which causes a larger spread in the pixel intensity distribution, like polarization.

I don't know how pyFAI handles everything. In my mind we can do one of two things:
 1. We can correct for the polarization ourselves and hand pyFAI `None` or `0` for the integration polarization correction, having already sorted it out for ourselves.
 1. We can copy the image, correct for polarization and mask the copied image and then integrate with a polarization factor passed to pyFAI using the original image and generated mask